perspective.el

```diff
@@ -466,6 +483,14 @@ perspective's local variables are set."
         (setq persp (persp-new name)))
       (run-hooks 'persp-before-switch-hook)
       (persp-activate persp)
+      ;; Only update the last switch time if one of the switching commands was
+      ;; explicitly called by the user.
+      (when (memq this-command '(persp-switch
+                                 persp-switch-last
+                                 persp-next
+                                 persp-prev
+                                 persp-switch-quick))
```

Comment:
I worry that this isn't friendly to users who are defining their own interactive perspective commands. It also doesn't include e.g. `persp-mode-line-click`. I'd just make this `setf` part of the behavior of `persp-switch`.

Reply:
Yes I was thinking about doing that, but what about `with-perspective`? Every time `with-perspective` is called, the perspective that was temporarily switched to would have a newer `last-switch-time` than the current perspective if we make the behavior part of `persp-switch`.

Maybe add a variable to temporarily disable setting the last switch time so the time doesn't get updated in `with-perspective` and possibly other places? Something like `persp-switch-norecord`?